subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemAttributesDescriber.java

```diff
@@ -130,6 +137,10 @@ public String describeMissingAttribute(Attribute<?> attribute, Object consumerVa
             sb.append("its usage (required ");
             describeUsage(consumerValue, sb);
             sb.append(")");
+        } else if (TargetJvmEnvironment.TARGET_JVM_ENVIRONMENT_ATTRIBUTE.equals(attribute)) {
+            sb.append("its target Java environment (preferred optimized for ");
```

Comment:
I would drop the preferred here

Reply:
This turns up in the error message a bout the producer. It says that the producer does not make a statement about the environment. The part in the `(...)` is then about what the consumer requested. I think leaving out the "preferred" make it very unclear that this part is about the consumer (and not the producer itself).

In other cases, we alway say "required". But that's unfitting, because his is really only a preference and no hard requirement.

Reply:
Full message

`- Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)`

vs

`- Doesn't say anything about its target Java environment (optimized for standard JVMs)`

Reply:
Ok, I see.